client/client.go

```diff
@@ -251,6 +251,8 @@ func defaultHTTPClient(hostURL *url.URL) (*http.Client, error) {
 	if err != nil {
 		return nil, err
 	}
+	transport.MaxIdleConns = 6
+	transport.IdleConnTimeout = 30 * time.Second
```

Comment:
Do we want these configurable on client opts like `WithMaxIdleCons` / `WithIdleConnTimeout` so it would be applied to client transport like we do for `WithTLSClientConfig` ?

Reply:
Maybe! I'll ping people for thoughts.

Regardless though, imo this change is about having better defaults – we can discuss/introduce options separately, unless this change breaks any existing client as-is.

Reply:
@tonistiigi @thaJeztah